src/Toolkit/Toolkit/GeoViewController.cs

```diff
@@ -0,0 +1,279 @@
+        #endregion Callouts
+
+#if MAUI
+        /// <summary>
+        /// Identifies the <see cref="GeoViewController"/> attached property.
+        /// </summary>
+        public static BindableProperty GeoViewControllerProperty =
+            BindableProperty.CreateAttached(
+                "GeoViewController",
+                typeof(GeoViewController),
+                typeof(GeoViewController),
+                null,
+                propertyChanged: OnGeoViewControllerChanged
+            );
+
+        private static void OnGeoViewControllerChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is GeoView geoView)
+            {
+                if (oldValue is GeoViewController controllerOld)
+                {
+                    controllerOld.DetachFromGeoView(geoView);
+                }
+
+                if (newValue is GeoViewController controllerNew)
+                {
+                    controllerNew.AttachToGeoView(geoView);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException("This property must be attached to a GeoView.");
+            }
+        }
+#else
+        /// <summary>
+        /// Identifies the <see cref="GeoViewController"/> attached property.
+        /// </summary>
+        public static readonly DependencyProperty GeoViewControllerProperty =
+            DependencyProperty.RegisterAttached(
+                "GeoViewController",
+                typeof(GeoViewController),
+                typeof(GeoViewController),
+                new PropertyMetadata(null, OnGeoViewControllerChanged)
+            );
+
+        /// <summary>
+        /// Gets the value of the <see cref="GeoViewController.GeoViewController"/> XAML attached property from the specified <see cref="GeoView"/>.
+        /// </summary>
+        /// <param name="geoView">The <see cref="GeoView"/> from which to read the property value.</param>
+        /// <returns>The value of the <see cref="GeoViewController.GeoViewController"/> XAML attached property on the target element.</returns>
+        public static GeoViewController? GetGeoViewController(GeoViewDPType geoView) => geoView.GetValue(GeoViewControllerProperty) as GeoViewController;
+
+        /// <summary>
+        /// Sets the value of the <see cref="GeoViewController.GeoViewController"/> XAML attached property on the specified object.
+        /// </summary>
+        /// <param name="geoView">The target <see cref="GeoView"/> on which to set the <see cref="GeoViewController.GeoViewController"/> XAML attached property.</param>
+        /// <param name="value">The property value to set.</param>
+        public static void SetGeoViewController(GeoViewDPType geoView, GeoViewController? value) => geoView.SetValue(GeoViewControllerProperty, value);
```

Comment:
Does `geoView` parameter need a null-check in these methods?  Maybe just something to throw ArgumentNullException instead of NRE.